pkg/apis/monitoring/v1alpha1/blackbox_exporter_types.go

```diff
@@ -0,0 +1,39 @@
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type BlackboxExporter struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec BlackboxExporterSpec `json:"spec"`
+}
+
+type BlackboxExporterSpec struct {
+	// Define resources requests and limits for each blackbox-exporter pod.
+	Resources v1.ResourceRequirements `json:"resources,omitempty"`
+	// Namespace for a blackbox-exporter deployment.
+	Namespace string `json:"namespace,omitempty"`
+	// Image to use for a blackbox-exporter deployment.
+	Image string `json:"image,omitempty"`
+	// Version string used for labeling objects
+	Version string `json:"version,omitempty"`
+	// +kubebuilder:
+	// Image of config-reloader sidecar container
+	ConfigmapReloaderImage string `json:"configmapReloaderImage,omitempty"`
+	// Secure listening port
+	Port int `json:"port,omitempty"`
+	// Insecure listening port
+	InternalPort int `json:"internalPort,omitempty"` // TODO: we should probably remove this
+	// Number of replicas of blackbox-exporter
+	Replicas int `json:"replicas,omitempty"`
+	// Configuration of blackbox-exporter in the same form as in
+	Modules map[string]string `json:"modules,omitempty"` // TODO: Consider using blackbox-exporter types for this
+	// Labels applied to all blackbox-exporter objects
+	CommonLabels map[string]string `json:"commonLabels,omitempty"`
```

Comment:
I feel like the hidden fields should probably not be exposed here. If people want to drop to this level, they should use jsonnet I think.

Reply:
agreed, I'll remove it